crates/turbopack-core/src/source_map/mod.rs

```diff
@@ -218,9 +218,9 @@ impl SourceMap {
                         )
                     })
                     .collect::<Vec<sourcemap::SourceMapSection>>();
-                Arc::new(CrateMapWrapper(DecodedMap::Index(SourceMapIndex::new(
-                    None, sections,
-                ))))
+                let map = SourceMapIndex::new(None, sections);
+                let map = map.flatten()?;
```

Comment:
I don't think we should flatten the source map in general.

Browsers support indexed source maps so we can use them here.
Our own lookup supports indexed source maps.

Only node.js doesn't support it, so we should only change the emitted SourceMaps to be flattened.